include/flatbuffers/grpc.h

```diff
@@ -287,8 +289,9 @@ template<class T> class SerializationTraits<flatbuffers::grpc::Message<T>> {
   }
 
   // Deserialize by pulling the
-  static grpc::Status Deserialize(grpc_byte_buffer *buffer,
+  static grpc::Status Deserialize(ByteBuffer *buf,
                                   flatbuffers::grpc::Message<T> *msg) {
+    grpc_byte_buffer *buffer = buf->c_buffer();
     if (!buffer) {
```

Comment:
The CI isn't happy about the casting to `c_buffer()`

```
external/com_github_grpc_grpc/include/grpcpp/impl/codegen/byte_buffer.h:167:21: error: 'grpc_byte_buffer* grpc::ByteBuffer::c_buffer()' is private
--
  | grpc_byte_buffer* c_buffer() { return buffer_; }
  | ^
  | In file included from tests/monster_test.grpc.fb.h:8:0,
  | from grpc/tests/grpctest.cpp:21:
  | bazel-out/k8-fastbuild/bin/_virtual_includes/flatbuffers/flatbuffers/grpc.h:294:45: error: within this context
  | grpc_byte_buffer *buffer = buf->c_buffer();
```

Reply:
The issue with this is the in the grpc header `byte_buffer.h` the only friend class for `SerializationTraits` is defined here https://github.com/grpc/grpc/blob/master/include/grpcpp/impl/codegen/byte_buffer.h#L159 and is `friend class SerializationTraits<ByteBuffer, void>;` 

In the case of `include/flatbuffers/grpc.h`, the first template parameter is `flatbuffers::grpc::Message<T>` and not `ByteBuffer`, so it is *not* a friend class.
Locally, I've had to change the grpc repo's header file to allow all `SerializationTraits` template specializations to be a friend class.
```  template <class T, class T1>
  friend class SerializationTraits;
```
So, this is really tough to solve unilaterally from just one repo and needs both the repos to be in agreement.

Reply:
@aardappel what do you think? Since I am no expert in cpp

Reply:
It is my fault regarding unfinished second patch option.
I suggest to forget about it for now.
Let's go back to original working casting one and see how to goes.
When we see a stable solution, let's try this one later.